### PR TITLE
Support ad domain services

### LINKED ISSUE #390: Azure Active Directory Domain Services - PQ3
Resource Support for Azure Active Directory Domain Services
Coverage:
1.Dev
2.Test
3.Documentation

### PR DESCRIPTION
**Pull request by sathish-progress**
### Description
support AD Domain Service(s)

### Issues Resolved
#390 

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [X] `rake lint` passes
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: docs/resources/azure_active_directory_domain_services.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+## Matchers
+
+This InSpec audit resource has the following special matchers. For a full list of available matchers, please visit our [Universal Matchers page](https://www.inspec.io/docs/reference/matchers/).
+
+### exists
+
+The control will pass if the filter returns at least one result. Use `should_not` if you expect zero matches.
+```ruby
+describe azure_active_directory_domain_services do
+  it { should exist }
```
Comment (soumyo13) [inline docs/resources/azure_active_directory_domain_services.md:95]: It should be "should_not"

File: docs/resources/azure_active_directory_domain_services.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+| is_admin_managed      | A list of admin managed configuration.                | `is_admin_managed`                           |
+| is_default            | A list of flags to indicate if it is default domains. | `is_default`                                 |
+| is_initial            | A list of flags to indicate if initial domains created by Microsoft Online Services.| `is_initial`   |
+| is_root               | A list of flags to indicate if verified root domain.  | `is_root`                                    |
+| is_verified           | A list of flags to indicate if the domain has completed domain ownership verification.| `is_verified`|
+| passwordNotificationWindowInDays | A list of password notification window days.| `passwordNotificationWindowInDays`          |
+| passwordValidityPeriodInDays | A list of password validity period in days.    |  `passwordValidityPeriodInDays`              |
+| supportedServices     | A list of capabilities assigned to the domain.        |  `supportedServices`                         |
+| state                 | A list of asynchronous operations scheduled.          |  `state`                                     |
```
Comment (soumyo13) [inline docs/resources/azure_active_directory_domain_services.md:60]: Why state is not pluralized?

**Review by IanMadd — APPROVED** [review]

File: docs/resources/azure_active_directory_domain_services.md (hunk 2 above)
Comment (sathish-progress) [reply docs/resources/azure_active_directory_domain_services.md:62]: table schema that was auto-generated based on the response for domain services .. 
Screenshot 2021-07-29 at 7 02 55 PM

**Review by ppradhan9 — APPROVED** [review]

**Review by sa-progress — APPROVED** [review]
LGTM

**Pull request merged**